lib/THC/THCTensorIndex.cu

```diff
@@ -245,6 +274,29 @@ void THCudaTensor_indexSelect(THCState *state, THCudaTensor *res_, THCudaTensor
   THCudaTensor_resize(state, res_, newSize, NULL);
   THLongStorage_free(newSize);
 
+  indices = THCudaTensor_newContiguous(state, indices);
+
+  if (THCudaTensor_isContiguous(state, src) && dim == 0)
```

Comment:
You should also check that `res_` is contiguous. `THCudaTensor_resize` doesn't guarantee that.